src/gpiocmds.c

```diff
@@ -172,32 +172,31 @@ soft_pwm_load_event(struct timer *timer)
 void
 command_config_soft_pwm_out(uint32_t *args)
 {
-    struct gpio_out pin = gpio_out_setup(args[1], !!args[3]);
+    struct gpio_out pin = gpio_out_setup(args[1], !!args[2]);
     struct soft_pwm_s *s = oid_alloc(args[0], command_config_soft_pwm_out
                                      , sizeof(*s));
     s->pin = pin;
-    s->cycle_time = args[2];
-    s->default_value = !!args[4];
-    s->max_duration = args[5];
+    s->default_value = !!args[3];
+    s->max_duration = args[4];
     s->flags = s->default_value ? SPF_ON : 0;
 }
 DECL_COMMAND(command_config_soft_pwm_out,
-             "config_soft_pwm_out oid=%c pin=%u cycle_ticks=%u value=%c"
+             "config_soft_pwm_out oid=%c pin=%u value=%c"
              " default_value=%c max_duration=%u");
 
 void
 command_schedule_soft_pwm_out(uint32_t *args)
 {
     struct soft_pwm_s *s = oid_lookup(args[0], command_config_soft_pwm_out);
-    uint32_t time = args[1], next_on_duration = args[2], next_off_duration;
+    uint32_t time = args[1], next_on_duration = args[2],
+        next_off_duration = args[3];
     uint8_t next_flags = SPF_CHECK_END | SPF_HAVE_NEXT;
-    if (next_on_duration == 0 || next_on_duration >= s->cycle_time) {
+    if (next_on_duration == 0) {
```

Comment:
This doesn't look correct to me - I think it should be:
```
if (next_on_duration == 0 || next_off_duration == 0)
```

Reply:
Honestly, I am not sure what this block does. But it seems to work as you suggest.

Reply:
That code block checks to see if there is actually PWM requested or full on/off.  In the full on/off case, there is no recurring timer scheduled, because we don't actually toggle the output in that case.

-Kevin

Reply:
Oh I understand. Thanks!